contentpal/src/main/java/org/dmfs/android/contentpal/predicates/BinaryPredicate.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.android.contentpal.predicates;
+
+import android.support.annotation.NonNull;
+
+import org.dmfs.android.contentpal.Predicate;
+import org.dmfs.iterables.ArrayIterable;
+import org.dmfs.iterables.decorators.Flattened;
+import org.dmfs.iterables.decorators.Mapped;
+import org.dmfs.iterators.Function;
+
+
+/**
+ * A {@link Predicate} which connects a number of given predicates with the given binary operator. Like this:
+ * <pre>{@code
+ * (predicate_1) operator (predicate_2) operator (predicate_3) … operator (predicate_n)O
```

Comment:
Is the "O" at the end of the line a typo here?

Reply:
Yes that was a typo. Fixed it.